commons-packet/commons-packet-manager/src/main/java/io/mosip/commons/packet/facade/PacketReader.java

```diff
@@ -53,13 +64,57 @@
     public String getField(String id, String field, String source, String process, boolean bypassCache) {
         LOGGER.info(PacketManagerLogger.SESSIONID, PacketManagerLogger.REGISTRATIONID, id,
                 "getFields for fields : " + field + " source : " + source + " process : " + process);
-        String value;
+        String value=null;;
+        if(source==null ||process==null||source.isBlank() || process.isBlank()) {
+    		JSONArray fieldJson=getSourceandProcess( field,  source,  process);
+    		if(fieldJson != null) {
+    			for(int i=0;i<fieldJson.length();i++) {
+    				try {
+    					if(source==null || source.isBlank()==true)
+						source=fieldJson.getString(i).split(",")[0].replace("source:", "");
+    					if(process==null || process.isBlank()==true)
+						process=fieldJson.getString(i).split(",")[1].replace("process:", "");
+						if(process.contains("|")) {
+							String[] processes=process.split("\\|");

```

Comment:
if inside if is the most confusing way of writing code. Please move some of this to proper functions.